cmd/metrics-v2.go

```diff
@@ -0,0 +1,1186 @@
+func getCacheSentBytesMD() MetricDescription {
+	return MetricDescription{
+		Namespace: minioNamespace,
+		Subsystem: cacheSubsystem,
+		Name:      sentBytes,
+		Help:      "Total number of bytes served from cache",
+		Type:      counterMetric,
+	}
+}
+func getHealObjectsTotalMD() MetricDescription {
+	return MetricDescription{
+		Namespace: healMetricNamespace,
+		Subsystem: objectsSubsystem,
+		Name:      total,
+		Help:      "Objects scanned in current self healing run",
+		Type:      gaugeMetric,
+	}
+}
+func getHealObjectsHealTotalMD() MetricDescription {
+	return MetricDescription{
+		Namespace: healMetricNamespace,
+		Subsystem: objectsSubsystem,
+		Name:      healTotal,
+		Help:      "Objects healed in current self healing run",
+		Type:      gaugeMetric,
+	}
+}
+func getHealObjectsFailTotalMD() MetricDescription {
+	return MetricDescription{
+		Namespace: healMetricNamespace,
+		Subsystem: objectsSubsystem,
+		Name:      errorsTotal,
+		Help:      "Objects for which healing failed in current self healing run",
+		Type:      gaugeMetric,
+	}
+}
+func getHealLastActivityTimeMD() MetricDescription {
+	return MetricDescription{
+		Namespace: healMetricNamespace,
+		Subsystem: timeSubsystem,
+		Name:      lastActivityTime,
+		Help:      "Time elapsed (in nano seconds) since last self healing activity. This is set to -1 until initial self heal activity",
+		Type:      gaugeMetric,
+	}
+}
+func getNodeOnlineTotalMD() MetricDescription {
+	return MetricDescription{
+		Namespace: clusterMetricNamespace,
+		Subsystem: nodesSubsystem,
+		Name:      onlineTotal,
+		Help:      "Total number of MinIO nodes online.",
+		Type:      gaugeMetric,
+	}
+}
+func getNodeOfflineTotalMD() MetricDescription {
+	return MetricDescription{
+		Namespace: clusterMetricNamespace,
+		Subsystem: nodesSubsystem,
+		Name:      offlineTotal,
+		Help:      "Total number of MinIO nodes offline.",
+		Type:      gaugeMetric,
+	}
+}
+func getMinIOVersionMD() MetricDescription {
+	return MetricDescription{
+		Namespace: minIOMetricNamespace,
+		Subsystem: softwareSubsystem,
+		Name:      versionInfo,
+		Help:      "MinIO Release tag for the server",
+		Type:      gaugeMetric,
+	}
+}
+func getMinIOCommitMD() MetricDescription {
+	return MetricDescription{
+		Namespace: minIOMetricNamespace,
+		Subsystem: softwareSubsystem,
+		Name:      commitInfo,
+		Help:      "Git commit hash for the MinIO release.",
+		Type:      gaugeMetric,
+	}
+}
+func getS3TTFBDistributionMD() MetricDescription {
+	return MetricDescription{
+		Namespace: s3MetricNamespace,
+		Subsystem: timeSubsystem,
+		Name:      ttfbDistribution,
+		Help:      "Distribution of the time to first byte across API calls.",
+		Type:      gaugeMetric,
+	}
+}
+
+func getS3TTFBMetric() MetricsGroup {
+	return MetricsGroup{
+		Metrics: []Metric{},
+		initialize: func(ctx context.Context, metrics *MetricsGroup) {
+
+			// Read prometheus metric on this channel
+			ch := make(chan prometheus.Metric)
+
+			wg := sync.WaitGroup{}
```

Comment:
```suggestion
			var wg sync.WaitGroup
```

Never allocate waitgroup